### DISCUSSION TITLE
[Core Protocol] <Question on "faultGameClockExtension": 10800(3hours),>

### DISCUSSION THREAD

**Discussion by AaronLee22** [ethereum-optimism/developers#625] [Core Protocol]
### Did you check the documentation and specs?

- [x] Yes

### Was there anything unclear or missing?

I understand that with FaultGameClockExtension, if a Challenger does not submit an attack claim within the CLOCK_EXTENSION time after the Proposer's initial submission, the Proposer's invalid output root will still be treated as valid.

When operating a small-scale L2, if FPS is run in a permissionless manner, any user can submit a proposal. In this case, the need for continuous monitoring and responding within 3 hours to avoid the absence of voluntary attacks or defenses feels burdensome. Additionally, if an invalid output root is used to attack a valid proposal and there is no response within the FaultGameClockExtension time, the attacker would win. Are there any measures in place to address this? For example, is there a system that automatically attacks invalid proposals?

If responding every 3 hours proves to be difficult, would it be better to run the dispute game in a permissioned manner? Or could the Guardian Role be used to prevent invalid proposals or withdrawals even after the FaultGameClockExtension time has passed, such as by deleting specific incorrect dispute games? -> As written on explainer docs "After a root is finalized through a game, an additional delay called the "airgap window" has been added before withdrawals can occur. During this period, the GUARDIAN role can reject the root." [link](https://docs.optimism.io/stack/fault-proofs/explainer)
<img width="1189" alt="Screenshot 2024-10-23 at 4 28 54 PM" src="https://github.com/user-attachments/assets/d3db16eb-0df2-426d-981c-cab404d00a8f">

Please help me out I would like to apply FPS on our network permissionlessly.

<img width="695" alt="image" src="https://github.com/user-attachments/assets/0bf8fc33-b940-4b6c-bc23-3dc74fe6f104">

### Did you check for duplicate questions?

- [X] Yes

### Please select the type of request

- [ ] Bug Report
- [ ] Feature Request

**Comment by imtipi** [comment]
https://github.com/ethereum-optimism/optimism/tree/develop/op-challenger
you can try op-challenger, which should automatically attacks invalid proposals

**Reply by sbvegan** [reply]
The clock extension time is not for the initial invalid output challenge. This can happen anytime within the challenge window. However, once the game to challenge an invalid out root, the game clock is used to ensure the game plays out in a timely manner. The op-challenger does this automatically and the op-dispute-mon service monitors these games. You can use these tools to automatically defend the chain. There is some nuance to running a challenger, so refer to our docs and file an issue if anything is not clear.

- https://github.com/ethereum-optimism/optimism/tree/develop/op-dispute-mon
- https://docs.optimism.io/stack/fault-proofs/challenger
